languages/bash.scm

```diff
@@ -22,23 +24,43 @@
   "fi"
   (string)
   (test_command)
+  ; TODO: etc.
 ] @prepend_space
 
 ;; Commands
 
-; One command per line, modulo many exceptions:
-; * Strings of && and ||
-; * Pipelines
-; * Subshells
-; * As arguments to if, while, for, etc.
-; * Others?...
+; NOTE "Command" is shorthand for a "unit of execution":
+; * Simple commands (e.g., binaries, builtins, functions, etc.)
+; * Command lists
+; * Command pipelines
 ;
-; FIXME Can this be generalised, or does *every* context need to be
-; individually enumerated?...
-(program (command) @prepend_spaced_softline @append_hardline)
-(if_statement _ "then" (command) @prepend_spaced_softline)
-(elif_clause _ "then" (command) @prepend_spaced_softline)
-(else_clause (command) @prepend_spaced_softline)
+; That is: [(command) (list) (pipeline)], per the grammar
+
+; FIXME I don't think it's possible to insert the necessary line
+; continuations; or, at least, it's not possible to insert them only in
+; a multi-line context. As such, all multi-line commands are forcibly
+; collapsed on to a single line for now... See Issue #172
+
+; One command per line in the following contexts:
+; * Top-level
+; * In any branch of a conditional
+; * <TODO: etc.>
+;
+; NOTE Because "command" is such a pervasive and general concept, each
+; context needs to be individually enumerated to account for exceptions.
+(program [(command) (list) (pipeline)] @prepend_hardline)
+(if_statement _ "then" [(command) (list) (pipeline)] @prepend_hardline)
+(elif_clause _ "then" [(command) (list) (pipeline)] @prepend_hardline)
+(else_clause [(command) (list) (pipeline)] @prepend_hardline)
+
+; Surround command list and pipeline delimiters with spaces
+(list ["&&" "||"] @append_space @prepend_space)
```

Comment:
Could we just put these in the list of symbols to surround with spaces? Do they really need to come within `list` and `pipeline`?

Reply:
I guess they don't. I've done it like this to make the context explicit, as I build the queries up incrementally; but you're right that it's redundant.

I'll add a note next to these rules (in #174) to ultimate generalise them into the list of symbols to surround with spaces.

Reply:
Note added in 6b4c69e8641193